stdlib/public/runtime/ReflectionMirror.mm

```diff
@@ -0,0 +1,790 @@
+}
+
+// func _getEnumCaseName<T>(_ value: T) -> UnsafePointer<CChar>?
+SWIFT_CC(swift) SWIFT_RUNTIME_STDLIB_INTERFACE
+const char *swift_EnumCaseName(OpaqueValue *value, const Metadata *T) {
+  return call(value, T, nullptr, [](ReflectionMirrorImpl *impl) { return impl->enumCaseName(); });
+}
+
+// func _opaqueSummary(_ metadata: Any.Type) -> UnsafePointer<CChar>?
+SWIFT_CC(swift) SWIFT_RUNTIME_STDLIB_INTERFACE
+const char *swift_OpaqueSummary(const Metadata *T) {
+  switch (T->getKind()) {
+    case MetadataKind::Class:
+    case MetadataKind::Struct:
+    case MetadataKind::Enum:
+    case MetadataKind::Optional:
+    case MetadataKind::Metatype:
+      return nullptr;
+    case MetadataKind::Opaque:
+      return "(Opaque Value)";
+    case MetadataKind::Tuple:
+      return "(Tuple)";
+    case MetadataKind::Function:
+      return "(Function)";
+    case MetadataKind::Existential:
+      return "(Existential)";
+    case MetadataKind::ObjCClassWrapper:
+      return "(Objective-C Class Wrapper)";
+    case MetadataKind::ExistentialMetatype:
+      return "(Existential Metatype)";
+    case MetadataKind::ForeignClass:
+      return "(Foreign Class)";
+    case MetadataKind::HeapLocalVariable:
+      return "(Heap Local Variable)";
+    case MetadataKind::HeapGenericLocalVariable:
+      return "(Heap Generic Local Variable)";
+    case MetadataKind::ErrorObject:
+      return "(ErrorType Object)";
+  }
+
+  swift_runtime_unreachable("Unhandled MetadataKind in switch.");
+}
+
+#if SWIFT_OBJC_INTEROP
+// func _getQuickLookObject<T>(_: T) -> AnyObject?
+SWIFT_CC(swift) SWIFT_RUNTIME_STDLIB_INTERFACE
+id swift_reflectionMirror_quickLookObject(OpaqueValue *value, const Metadata *T) {
+  return call(value, T, nullptr, [](ReflectionMirrorImpl *impl) { return impl->quickLookObject(); });
+}
+#endif
+
+// NB: This function is not used directly in the Swift codebase, but is
+// exported for Xcode support and is used by the sanitizers. Please coordinate
+// before changing.
+//
+/// Demangles a Swift symbol name.
+///
+/// \param mangledName is the symbol name that needs to be demangled.
+/// \param mangledNameLength is the length of the string that should be
+/// demangled.
+/// \param outputBuffer is the user provided buffer where the demangled name
+/// will be placed. If nullptr, a new buffer will be malloced. In that case,
+/// the user of this API is responsible for freeing the returned buffer.
+/// \param outputBufferSize is the size of the output buffer. If the demangled
+/// name does not fit into the outputBuffer, the output will be truncated and
+/// the size will be updated, indicating how large the buffer should be.
+/// \param flags can be used to select the demangling style. TODO: We should
+//// define what these will be.
+/// \returns the demangled name. Returns nullptr if the input String is not a
+/// Swift mangled name.
+SWIFT_RUNTIME_EXPORT
+char *swift_demangle(const char *mangledName,
```

Comment:
It's so weird to have this function in this file. `swift_demangle` is a reasonable entrypoint to export from the runtime, but it's not specific to reflection at all. Maybe we can move it elsewhere?

Reply:
Agreed. It's there pretty much because it used to be in Reflection.mm and I mindlessly copied it over. I've now moved it to Demangle.cpp, which seems more appropriate.